src/playercards/cards/KohakuNarukami.ttslua

```diff
@@ -134,14 +148,45 @@ function setUiState(params)
 end
 
 function errorMessage(_, _, triggeringButton)
+  local reactionToken = getReactionToken()
+  local actionSpent = false
+  if reactionToken ~= nil then actionSpent = reactionToken.is_face_down end
+
   local numInBag = blessCurseManagerApi.getBlessCurseInBag()
-  if triggeringButton == "Action" then
+  if actionSpent and triggeringButton ~= "ElderSign" then
+    broadcastToAll("Ability already used.", "Red")
+  elseif triggeringButton == "Action" then
     broadcastToAll("There are not enough Blesses and/or Curses in the chaos bag.", "Red")
   elseif numInBag.Bless == 10 and numInBag.Curse == 10 then
     broadcastToAll("No more tokens can be added to the chaos bag.", "Red")
   elseif numInBag.Bless < numInBag.Curse then
-    broadcastToAll("There are more Bless tokens than Curse tokens in the chaos bag.", "Red")
-  else
     broadcastToAll("There are more Curse tokens than Bless tokens in the chaos bag.", "Red")
+  else
+    broadcastToAll("There are more Bless tokens than Curse tokens in the chaos bag.", "Red")
   end
 end
+
+-- get reaction token on investigator card
+function getReactionToken()
+  local reactionToken
+  local matColor = playermatApi.getMatColorByPosition(self.getPosition())
+  local pos = zones.getZonePosition(matColor, "Investigator")
+  local searchResult = searchLib.atPosition(pos, "isCardOrDeck")
+  if #searchResult == 1 then
+    local reaction = searchLib.onObject(searchResult[1], "isTileOrToken")
+    if #reaction == 1 then
+      if string.sub(reaction[1].getData().Nickname, -8, -1) == "Reaction" then
+        reactionToken = reaction[1]
+      end
+    end
+  end
```

Comment:
```suggestion
  local rot = playermatApi.returnRotation(matColor)
  local size = Vector(2.3, 2, 3.4)
  local searchResult = searchLib.inArea(pos, rot, , "isUniversalToken")
  for _, obj in ipairs(searchResult) do
      if string.sub(obj.getName(), -8, -1) == "Reaction" then
        reactionToken = obj
        break
      end
  end
```
Untested, but something like this would be better to get the ability token imo (one search instead of 2)